community/kernel/src/main/java/org/neo4j/storageengine/api/StorageStatement.java

```diff
@@ -33,20 +33,28 @@
  * are valuable to reuse over a reasonably large window to reduce garbage churn in general.
  *
  * A {@link StorageStatement} must be {@link #acquire() acquired} before use. After use the statement
- * should be {@link #close() closed}. After closed the statement can be acquired again.
+ * should be {@link #release() released}. After released the statement can be acquired again.
+ * Creating and closing {@link StorageStatement} and there's also benefits keeping these statements opened
+ * during a longer period of time, with the assumption that it's still one thread at a time using each.
+ * With that in mind these statements should not be opened and closed for each operation, perhaps not even
+ * for each transaction.
```

Comment:
Yes, a quick reminder that we are below the transaction state in the stack would be good, I think.